cmd/gateway-handlers.go

```diff
@@ -246,6 +246,13 @@ func (api gatewayAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Re
 		}
 	}
 
+	// Make sure we hex encode md5sum here.
+	metadata["etag"] = hex.EncodeToString(md5Bytes)
+
+	// Make sure we save `Content-Md5` right here, backend callers might use
+	// this to send it back to the backend implementations.
+	metadata["content-md5"] = r.Header.Get("Content-Md5")
```

Comment:
thisis not needed. i.e at manta's PutObject() it can check `data.md5sum`

Reply:
@stack72 @cheapRoc can you fix these as well?

Reply:
can these entries be removed @cheapRoc looks like these are still around ..

Reply:
I don't really know what these are performing. Any context/idea before I remove them?

Reply:
It was precursor change to passing content-md5 directly to the remote backend for validation.  But it is not needed in this PR.